fix(apple): App hanging when Watchdog termination is enabled (#7681)
getsentry/sentry-docs: src/platforms/apple/common/configuration/app-hangs.mdx

The SDK reports an app hang immediately, but doesn’t report the exact duration because the [watchdog](https://developer.apple.com/documentation/xcode/addressing-watchdog-terminations) could kill the app anytime if it's blocking the main thread.

> When watchdog termination tracking is enabled, the app hang code runs in the background, even if `enableAppHangTracking` is disabled, and won’t report app hangs. This is done to minimize the number of false errors being reported by watchdog termination tracking, (such as if the OS kills an app causing an app hang).

Reviewer: ```suggestion
When watchdog termination tracking is enabled, the app hang code runs in the background, even if `enableAppHangTracking` is disabled, and won’t report app hangs. This minimizes the number of false errors reported by watchdog termination tracking, for example, if the OS kills an app causing an app hang.
```
